crates/ty_completion_eval/src/main.rs

```diff
@@ -0,0 +1,588 @@
+    match args.command {
+        Command::ShowOne(ref cmd) => {
+            let Some(source) = sources
+                .iter()
+                .find(|source| cmd.matches_source_task(source))
+            else {
+                anyhow::bail!("could not find task named `{}`", cmd.task_name);
+            };
+            let tasks = source.to_tasks(&tmp_eval_dir)?;
+            let matching: Vec<&Task> = tasks.iter().filter(|task| cmd.matches_task(task)).collect();
+            anyhow::ensure!(
+                !matching.is_empty(),
+                "could not find any tasks matching the given criteria",
+            );
+            anyhow::ensure!(
+                matching.len() < 2,
+                "found more than one task matching the given criteria",
+            );
+            let task = &matching[0];
+            let completions = task.completions()?;
+
+            let mut stdout = std::io::stdout().lock();
+            for (i, c) in completions.iter().enumerate() {
+                write!(stdout, "{}", c.name.as_str())?;
+                if let Some(module_name) = c.module_name {
+                    write!(stdout, " (module: {module_name})")?;
+                }
+                if task.cursor.answer.matches(c) {
+                    write!(stdout, " (*, {}/{})", i + 1, completions.len())?;
+                }
+                writeln!(stdout)?;
+            }
+            writeln!(stdout, "-----")?;
+            writeln!(stdout, "found {} completions", completions.len())?;
+            Ok(ExitCode::SUCCESS)
+        }
+        Command::All(AllCommand { threshold, tasks }) => {
+            let mut precision_sum = 0.0;
+            let mut results_wtr = None;
+            if let Some(ref tasks) = tasks {
+                let mut wtr = csv::Writer::from_path(SystemPath::new(tasks))?;
+                wtr.serialize(("name", "file", "index", "rank"))?;
+                results_wtr = Some(wtr);
+            }
+            for source in &sources {
+                for task in source.to_tasks(&tmp_eval_dir)? {
+                    let completions = task.completions()?;
+                    let rank = task.rank(&completions)?;
+                    precision_sum += rank.map(|rank| 1.0 / f64::from(rank)).unwrap_or(0.0);
+                    if let Some(ref mut wtr) = results_wtr {
+                        wtr.serialize((
+                            &task.name,
+                            &task.cursor.path.as_str(),
+                            task.cursor.index,
+                            rank,
+                        ))?;
+                    }
+                }
+            }
+            #[allow(clippy::cast_precision_loss)]
+            let mrr = precision_sum / (sources.len() as f64);
+            if let Some(ref mut wtr) = results_wtr {
+                wtr.flush()?;
+            }
+
+            let mut out = std::io::stdout().lock();
+            writeln!(out, "mean reciprocal rank: {mrr}")?;
```

Comment:
The inner physicist in me always gets slightly nervous if statistical measures are presented to full numerical precision (*"mean reciprocal rank: 0.20409790112917506"*) :smile:. The actual precision of the MRR measure could be estimated by computing the standard deviation alongside the mean, but that obviously seems like overkill here. Given that individual reciprocal ranks can have values like 1/100 = 0.01, and given that we only have ~10 tasks, three or four decimal digits seem more than enough, so maybe
```suggestion
            writeln!(out, "mean reciprocal rank: {mrr:.4}")?;
```
(Due to the large spread in reciprocal ranks and the small number of tasks, the actual standard deviation is around 0.34 at the moment, higher than the MRR. That would mean that even one digit is enough)